File: Golang/Squares of a Sorted Array.go
func sortedSquares(nums []int) []int {
    ans := make([]int,len(nums))
    start := 0
    end := len(nums) - 1
    
    for i:=len(nums)-1;i>=0;i--{
        if abs(nums[start]) > abs(nums[end]) {
            ans[i] = nums[start] * nums[start]
            start++
        } else {
            ans[i] = nums[end] * nums[end]
            end--
        }
    }
    return ans
}

func abs(a int) int {
    if a < 0 {
        return -a
    }
    return a
}

File: Python3/Squares of a Sorted Array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        start = 0
        end = len(nums) - 1
        ans = [0] * len(nums)

        for i in range(len(nums)-1,-1,-1):
            if abs(nums[start]) > abs(nums[end]):
                ans[i] = nums[start] * nums[start]
                start = start + 1
            else:
                ans[i] = nums[end] * nums[end]
                end = end - 1
        return ans